### PR TITLE
fix/refactor-docs

### PR DESCRIPTION
**Pull request by saibotma**

### CODE REVIEW

File: lib/src/exception/not_defined_exception.dart [hunk 1]
```diff
@@ -1,34 +1,15 @@
...
- *
- *  - Did you call your register method?
- *
- *    Maybe you missed to call the register function!
- *
- *    Be sure that you call:
- *    - Injector().registerDependency<IInterface>((_) => Implementation());
- *
- */
+/// Gets thrown when trying get a dependency that has not been registered yet.
```
Comment (JulianBissekkou) [inline lib/src/exception/not_defined_exception.dart]: should be 
`..when trying TO get a dependency...`

File: lib/src/injector_base.dart [hunk 2]
```diff
@@ -6,49 +6,45 @@ import 'package:injector/src/factory/provider_factory.dart';
...
-  /**
-   * The Map that contains:
-   *
-   *  [SingletonFactory] - The single instances
-   *
-   * [ProviderFactory] - The provider factory that contains the type information.
-   *                     If creates a new instance every time
-   */
+  /// Stores [SingletonFactory] and [ProviderFactory] instances that have been
+  /// registered by [registerSingleton] and [registerDependency] respectively.
```
Comment (JulianBissekkou) [inline lib/src/injector_base.dart]: Do we expose `Factory` and a way to register custom factories? If so we should mention it here :)

File: lib/src/injector_base.dart [hunk 3]
```diff
@@ -6,49 +6,45 @@ import 'package:injector/src/factory/provider_factory.dart';
...
+  /// The signature of a dependency consists of [T]
+  /// and the optional [dependencyName].
+  ///
+  /// The injector will always create new instances of dependencies that got
+  /// registered this way when calling [getDependency].
+  ///
+  /// Example:
+  /// ```dart
+  /// abstract class UserService {
+  ///     void Login(String username, String password);
```
Comment (JulianBissekkou) [inline lib/src/injector_base.dart]: the function name should be lowercase here :S

File: lib/src/injector_base.dart [hunk 4]
```diff
@@ -6,49 +6,45 @@ import 'package:injector/src/factory/provider_factory.dart';
...
+  /// }
+  ///
+  /// class UserServiceImpl implements UserService {
+  ///     void Login(String username, String password) {
```
Comment (JulianBissekkou) [inline lib/src/injector_base.dart]: here as well :)

File: lib/src/injector_base.dart [hunk 5]
```diff
@@ -6,49 +6,45 @@ import 'package:injector/src/factory/provider_factory.dart';
...
+  ///         // does something..
+  ///     }
+  /// }
+  ///
+  /// Injector.appInstance.registerDependency<UserService>((_) => new UserService);
```
Comment (stefanschaller) [inline lib/src/injector_base.dart]: you need the brackets :o `()`
`new UserServiceImpl()`

You can not create an instance of an abstract class

File: lib/src/injector_base.dart [hunk 6]
```diff
@@ -62,30 +58,36 @@ class Injector {
...
+  /// }
+  ///
+  /// class UserServiceImpl implements UserService {
+  ///  void Login(String username, String password){
+  ///    .....
+  ///    .....
+  ///  }
+  /// }
+  ///
+  /// Injector.appInstance.registerSingleton<UserService>((_) => new UserService);
```
Comment (stefanschaller) [inline lib/src/injector_base.dart]: some :D 
#StefanTheCompiler

**Comment by stefanschaller** [comment]
Don't forget to increase the version number :D

File: lib/src/injector_base.dart (hunk 2 above)
Comment (saibotma) [reply lib/src/injector_base.dart]: I´ve removed the `Factory` stuff from the docs altogether.

**Pull request merged**